lib/kamal/commands/builder/pack.rb

```diff
@@ -0,0 +1,26 @@
+class Kamal::Commands::Builder::Pack < Kamal::Commands::Builder::Base
+  def push
+    combine \
+      pack(:build,
+        config.repository,
+        "--platform", platform,
+        "--builder", pack_builder,
+        buildpacks,
+        "-t", config.absolute_image,
+        "-t", config.latest_image,
+        "--env", "BP_IMAGE_LABELS=service=#{config.service}",
+        secrets.map { |secret| [ "--env", Kamal::Utils.sensitive(ENV[secret]) ] },
```

Comment:
Check how this is going to work with the new secrets PR https://github.com/basecamp/kamal/pull/924